File: python/bench_sorts.py
#!/usr/bin/python
import random

from native_sort import unsorted
from heap_sort import heap_sort
from merge_sort import merge_sort
from quick_sort import quick_sort
from quick_sort2 import quick_sort2

def bench_sort(tab, func):
    return func(tab)

for tab in (unsorted,):
    for sort_algo in (sorted, merge_sort, quick_sort, quick_sort2, heap_sort):
        bench_sort(tab, sort_algo)

File: go/bench_sorts.go
package main

import (
	"fmt"
	"flag"
	"intslice"
	"sorting"
)

func main() {
	async := flag.Bool("async", false, "launch asynchronous sorting algos")
	sync := flag.Bool("sync", false, "launch synchronous sorting algos")
	native := flag.Bool("native", false, "launch the native sort")
	merge := flag.Bool("merge", false, "launch the merge sort")
	quick := flag.Bool("quick", false, "launch the quick sort")
	quick2 := flag.Bool("quick2", false, "launch the quick sort in-place")
	flag.Parse()
	sorts := map[*bool] func (intslice.IntSlice, bool) intslice.IntSlice {
	        native: sorting.NativeSort,
		merge: sorting.MergeSort,
		quick: sorting.QuickSort,
		quick2: sorting.QuickSort2,
	}
	for b, f := range sorts {
		if *b && *async == true {
			unsorted := intslice.Perm(1000)
			fmt.Println(f(unsorted, true))
		}
		if *b && (*sync == true || *async == false) {
			unsorted := intslice.Perm(1000)
			fmt.Println(f(unsorted, false))
		}
	}
}
